components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java

```diff
@@ -500,4 +600,156 @@ public void testSetCookieExistCookieConfig() {
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie storedCookie = capturedCookies.get(0);
+
+        assertEquals(storedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(storedCookie.getPath(), "Dummy-Path");
+        assertEquals(storedCookie.getMaxAge(), age);
+    }
+
+    @Test
+    public void testGetCookieExistingCookie() {
+
+        Cookie[] cookies = getAuthenticationCookies();
+        when(request.getCookies()).thenReturn(cookies);
+
+        Cookie out = FrameworkUtils.getCookie(request, cookies[0].getName());
+
+        assertEquals(out, cookies[0]);
+    }
+
+    @Test
+    public void testGetCookieNonExistingCookie() {
+
+        Cookie out = FrameworkUtils.getCookie(request, "nonExistingCookie");
+        assertNull(out);
+    }
+
+    @Test
+    public void testGetHashOfCookie() {
+
+        Cookie cookie = new Cookie(FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue");
+
+        String out = FrameworkUtils.getHashOfCookie(cookie);
+
+        assertEquals(out, DigestUtils.sha256Hex("commonAuthIdValue"));
+    }
+
+    @Test
+    public void testGetPasswordProvisioningUIUrlEmptyURL() {
+
+        String out = FrameworkUtils.getPasswordProvisioningUIUrl();
+        assertEquals(out, FrameworkConstants.SIGN_UP_ENDPOINT);
+    }
+
+    @Test
+    public void testGetPasswordProvisioningUIUrlNonEmptyURL() {
+
+        String dummyURL = "Dummy_Provisioning_URL";
+        IdentityConfigParser.getInstance().getConfiguration().put("JITProvisioning.PasswordProvisioningUI", dummyURL);
+        String out = FrameworkUtils.getPasswordProvisioningUIUrl();
+        assertEquals(out, "/accountrecoveryendpoint/signup.do");
+    }
+
+    @Test
+    public void testGetUserNameProvisioningUIUrlEmptyURL() {
+
+        String out = FrameworkUtils.getUserNameProvisioningUIUrl();
+        assertEquals(out, FrameworkConstants.REGISTRATION_ENDPOINT);
+    }
+
+    @Test
+    public void testGetUserNameProvisioningUIUrlNonEmptyURL() {
+
+        String dummyURL = "Dummy_Provisioning_URL";
+        IdentityConfigParser.getInstance().getConfiguration().put("JITProvisioning.UserNameProvisioningUI", dummyURL);
+
+        String out = FrameworkUtils.getUserNameProvisioningUIUrl();
+        assertEquals(out, "/accountrecoveryendpoint/register.do");
+    }
+
+    private void setMockedSessionContextCache() {
+
+        mockStatic(SessionContextCache.class);
+        when(SessionContextCache.getInstance()).thenReturn(mockedSessionContextCache);
+    }
+
+    @Test
+    public void testGetSessionContextFromCacheNullCacheEntry() {
+
+        setMockedSessionContextCache();
+
+        SessionContext out = FrameworkUtils.getSessionContextFromCache(DUMMY_CACHE_KEY);
+        System.out.println(out);
```

Comment:
Let's remove the sys.out

Reply:
Addressed with commit 387514e50f42aead307ee0f529bb17b806672a14